core/src/main/scala/org/apache/spark/rpc/akka/AkkaRpcEnv.scala

```diff
@@ -237,6 +237,12 @@ private[spark] class AkkaRpcEnv private[akka] (
   }
 
   override def toString: String = s"${getClass.getSimpleName}($actorSystem)"
+
+  def deserialize[T](deserializationAction: () => T): T = {
+    JavaSerializer.currentSystem.withValue(actorSystem.asInstanceOf[ExtendedActorSystem]) {
```

Comment:
> but why do we need the actor system to deserialize it? Can't we just deserialize it with JavaSerializer?

Oh, that's because `WorkerInfo` and `ApplicationInfo` contain a reference to RpcEndpointRef.